### PR TITLE
Tracking receipts

### PR DESCRIPTION
**Pull request by malinajirka**
Parent issue #4039 

Merge instructions
1. Make sure https://github.com/woocommerce/woocommerce-android/pull/4244 is merged [x]
2. Remove "do not merge" label
3. Merge this PR

This PR adds tracking to send/print receipt functionality.
receipt_email_tapped: The user tapped on the button to email a receipt.
receipt_email_failed: Email application not found in the system.

receipt_print_tapped: The user tapped on the button to print a receipt.
receipt_print_failed: Printing the receipt failed.
receipt_print_canceled: The user canceled printing the receipt.
receipt_print_success: The print job was passed to the OS.

Note 1:
The iOS app tracks also 
receipt_email_canceled: The user canceled sending the receipt by email.
receipt_email_success: The receipt was sent by email.
However, the Android OS doesn't inform the source app about results of implicit intents, so these events are not used on Android.

Note 2:
Tracking print result - PrintManager doesn't provide a public callback which would inform the caller about the result of a print job. The current status of a PrintJob is available on the PrintJob itself. We could query the PrintJob every few seconds to find out the current status and track the events. However, implementing it would not be trivial. Therefore my suggestion is to track the current status of the print job when the WCAndroid comes back to the foreground from the OS print UI. If the job is enqueued/started/completed/blocked we track `success` event, if it's failed we track `failed` event (even though the user can still retry), if it's canceled we track `cancelled` event. This logic should be pretty close to how the iOS app behaves.

To test:
Verify that the events are tracked - you can use "print to pdf" if you don't have a printer available. I'm not sure how to simulated "Print failed" state, but since it shares the same codepath with success/cancelled I don't think we need to test it.

Update release notes:

- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WooCommerce/src/main/kotlin/com/woocommerce/android/analytics/AnalyticsTracker.kt [hunk 1]
```diff
@@ -234,7 +234,6 @@ class AnalyticsTracker private constructor(private val context: Context) {
         CARD_PRESENT_COLLECT_PAYMENT_FAILED,
         CARD_PRESENT_COLLECT_PAYMENT_SUCCESS,
 
-        // -- Card Present Payments - connection
```
Comment (kidinov) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/analytics/AnalyticsTracker.kt:237]: I think having a comment per section is nice. Ideally even to use regions, but we don't use them here

https://www.jetbrains.com/help/idea/code-folding-settings.html

**Review by kidinov — CHANGES_REQUESTED** [review]
All looks good. Just answer if you are ok to group events with comments (check Products)

File: WooCommerce/src/main/kotlin/com/woocommerce/android/analytics/AnalyticsTracker.kt (hunk 1 above)
Comment (malinajirka) [reply WooCommerce/src/main/kotlin/com/woocommerce/android/analytics/AnalyticsTracker.kt:237]: I like regions and I used to use them extensively - even in code. For this particular case, I think we should either introduce regions to all the keys in this file or not introduce them. I'll add it to my nice-to-have todo list.

Comment (malinajirka) [reply WooCommerce/src/main/kotlin/com/woocommerce/android/analytics/AnalyticsTracker.kt:237]: Ohh I understand your comment now - I didn't intend to remove this comment. I'll revert this change.

**Review by kidinov — APPROVED** [review]

**Pull request merged**